gems/aws-crt/lib/aws-crt/http/headers.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module Aws
+  module Crt
+    # High level Ruby abstractions for CRT HTTP functionality
+    module Http
+      # HTTP Headers
+      class Headers
+        include Aws::Crt::ManagedNative
+        native_destroy Aws::Crt::Native.method(
+          :http_headers_release
+        )
+
+        def initialize(headers = {})
+          blob = StringBlob.encode(headers.each_pair { |k, v| [k, v] }.flatten)
```

Comment:
I think just `headers.flatten` gives you the same behavior.

Reply:
It doesn't quite - the blob is a flat list, eg: 
```ruby
h = ["h1", "v1", "h2", "v2"]
```
Using flatten on that doesn't do anything:
```ruby
["h1", "v1", "h2", "v2"].flatten
=> ["h1", "v1", "h2", "v2"]
```

Reply:
I was testing with
```
[3] pry(Aws)> headers = { a: 'b', c: 'd' }
=> {:a=>"b", :c=>"d"}
[4] pry(Aws)> headers.each_pair { |k, v| [k, v] }.flatten
=> [:a, "b", :c, "d"]
[5] pry(Aws)> headers.flatten
=> [:a, "b", :c, "d"]
```

Reply:
Ah, my mistake - that does work here.  I was thinking about the other direction (array to hash).  I've updated in message and headers.